src/client/java/daos-java-api/src/main/native/com_intel_daos_client_DaosFsClient.c

```diff
@@ -0,0 +1,911 @@
+	if (parent_handle) dfs_release(parent_handle);
+	return rc;
+}
+
+JNIEXPORT void JNICALL Java_com_intel_daos_client_DaosFsClient_mkdir
+  (JNIEnv *env, jobject client, jlong dfsPtr, jstring path, jint mode, jboolean recursive){
+	const char* path_str = (*env)->GetStringUTFChars(env, path, NULL);
+	dfs_t *dfs = *(dfs_t**)&dfsPtr;
+	char *dirs, *bases, *parent_dir, *base;
+	dfs_obj_t *parent_handle = NULL;
+	mode_t parent_mode;
+
+	char *parentError = NULL;
+
+	dirs = strdup(path_str);
+	parent_dir = dirname(dirs);
+	bases = strdup(path_str);
+	base = basename(bases);
+	int rc = 0;
+	if ((strlen(parent_dir) > 0) && (strcmp(parent_dir, "/") != 0)){
+		parentError = (char*)malloc(ERROR_PATH_LEN);
+		parentError[0] = '\0';
+		rc = mkdirs(dfs, parent_dir, mode, recursive, &parent_handle, parentError);
+	}
+	if (rc) {
+		char *dir_msg = parent_dir;
+		char *tmp;
+		if(recursive){
+			tmp = "Failed to create parent or ancestor directories (%s)";
+			if (parentError[0] != '\0') {
+				dir_msg = parentError;
+			}
+		} else {
+			tmp = "Parent directory doesn't exist (%s)";
+		}
+		char *msg = (char *)malloc(strlen(tmp) + strlen(dir_msg));
+		sprintf(msg, tmp, dir_msg);
+		throw_exception(env, msg, rc);
+	} else {
+		rc = dfs_mkdir(dfs, parent_handle, base, mode);
+		if (rc) {
+			char *tmp = "Failed to create directory (%s) under parent directory (%s)";
+			char *msg = (char *)malloc(strlen(tmp) + strlen(base) + strlen(parent_dir));
+			sprintf(msg, tmp, base, parent_dir);
+			throw_exception(env, msg, rc);
+		}
+	}
+	if (dirs) free(dirs);
+	if (bases) free(bases);
+	if (parentError) free(parentError);
+	if (parent_handle) dfs_release(parent_handle);
+	(*env)->ReleaseStringUTFChars(env, path, path_str);
+}
+
+JNIEXPORT jlong JNICALL Java_com_intel_daos_client_DaosFsClient_createNewFile
+  (JNIEnv *env, jobject client, jlong dfsPtr, jstring parentPath, jstring name,
+		  jint mode, jint accessFlags, jint objectId, jint chunkSize, jboolean createParent){
+	dfs_t *dfs = *(dfs_t**)&dfsPtr;
+	const char* parent_path = (*env)->GetStringUTFChars(env, parentPath, NULL);
+	const char* file_name = (*env)->GetStringUTFChars(env, name, NULL);
+	char *parentError = NULL;
+	dfs_obj_t *file = NULL, *parent = NULL;
+	mode_t tmp_mode;
+	int rc = dfs_lookup(dfs, parent_path, O_RDWR, &parent, &tmp_mode, NULL);
+	if (rc) {
+		if (createParent) {
+			parentError = (char*)malloc(ERROR_PATH_LEN);
+			parentError[0] = '\0';
+			rc = mkdirs(dfs, parent_path, mode, 1, &parent, parentError);
+			if (rc) {
+				char *dir_msg = parent_path;
+				if (parentError[0] != '\0') {
+					dir_msg = parentError;
+				}
+				char *tmp = "Failed to create parent/ancestor directories (%s)";
+				char *msg = (char *)malloc(strlen(tmp) + strlen(dir_msg));
+				sprintf(msg, tmp, dir_msg);
+				throw_exception(env, msg, rc);
+				goto out;
+			}
+		} else {
+			char *tmp = "Failed to find parent directory (%s)";
+			char *msg = (char *)malloc(strlen(tmp) + strlen(parent_path));
```

Comment:
(style)  line over 80 characters

Reply:
just reformatted code and added more documentation. please review